library/md5.c

```
@@ -397,6 +428,12 @@ int mbedtls_md5_self_test( int verbose )
         mbedtls_printf( "\n" );
 
     return( 0 );
+
+fail:
+    if( verbose != 0 )
+        mbedtls_printf( "failed\n" );
+
+    return( 1 );
 }
```

We should return the error code from `mbedtls_md5_ext` here if it fails.